### PR TITLE
Add recipe for Gensim

### PR DESCRIPTION
**Pull request by souravsingh**

### CODE REVIEW

**Comment by souravsingh** [comment]
Pinging @tmylk for the Gensim recipe.

**Comment by tmylk** [comment]
Hi @souravsingh Thank you for creating it. Is there a way to actually run gensim tests after building the recipe?

**Comment by souravsingh** [comment]
@tmylk I believe that gensim tests can be added in the feedstock after the recipe is finished building. I would like to know from the conda-forge devs if this is possible.

**Comment by souravsingh** [comment]
Hmm... The Travis build is timing out. Would restarting the build fix the issue?

**Comment by souravsingh** [comment]
The travis build is Failing due to a timeout. @synapticarbors Is there a solution to fix it?

**Comment by tmylk** [comment]
@souravsingh When you test it on your linux machine, does this command work `nosetests --exe -v gensim`?

In Travis it hangs:
```
Test doc2vec parallel training. ... 
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

**Comment by souravsingh** [comment]
@tmylk I checked out the tests in Linux(Ubuntu 16.04) and all the tests passed(8 skipped). I am inclined to think this is a problem with Travis.

**Comment by souravsingh** [comment]
Can someone help in rectifying the problem with Travis build?

**Comment by tmylk** [comment]
Waiting for the travis to be re-run to resolve this sporadic fail

**Comment by jakirkham** [comment]
Restarted. Also have one question. Otherwise it is fine.

File: recipes/gensim/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - numpy >=1.3
+    - pyro4 >=4.27
```
Comment (souravsingh) [inline recipes/gensim/meta.yaml]: I believe they are indeed needed.

Comment (tmylk) [inline recipes/gensim/meta.yaml]: I think only setuotools is needed. It was same case with smart_open recipe

**Comment by souravsingh** [comment]
@tmylk Does tests for Gensim fail in Mac OS X?

**Comment by tmylk** [comment]
@souravsingh Gensim wheel builds without any failures on OSX, except Python 3.3
See https://travis-ci.org/MacPython/gensim-wheels

**Comment by souravsingh** [comment]
Strange, the tests shouldn't have any reason to error in Travis when it passes in Circle and Appveyor. I suggest skipping Travis builds, what do you think @tmylk

**Comment by tmylk** [comment]
This test always passes without any problems in the main gensim repo. "Test doc2vec parallel training. ... " @souravsingh Does it pass the conda recipe tests on your machine?

**Comment by souravsingh** [comment]
@tmylk I have tested the Gensim recipe in Linux and Windows and it runs nicely. I do not have access to Mac to test the recipe for Mac OS X.

**Comment by tmylk** [comment]
Failing to reproduce the hanging tests locally. Is it a Travis artefact?

**Comment by souravsingh** [comment]
@tmylk Do we need to run nosetests in the recipe? I had a discussion with the devs at conda-forge and they said it won't be possible to run full test suite on CI due to the restricted nature of the machines.

**Comment by tmylk** [comment]
how else do we know that the resulting package is correct?
we run nose on Travis in gensim github repo and it runs ok.

**Comment by jakirkham** [comment]
It should be ok to run tests. The only question is how intensive those tests are. The resources here are shared amongst all packages. So if it is taking an hour to run the full test suite for example, that might be a bit extreme and we would need to discuss alternatives. However, if we are talking a few minutes, sure one could run the whole test suite and that would be fine.

Ideally the code that is released is passing its test suite anyways. The point of tests here is not to replace CI for the code base or similar. Instead they exist to provide basic checks like did things link ok, could a user do basic things with this package, did we miss some dependencies, are we getting numerical errors from compiled functions, etc. Does that make sense?

**Comment by jakirkham** [comment]
Toggling to get CircleCI running.

**Comment by tmylk** [comment]
@jakirkham Thanks. Completely agree. The tests take 17 mins. They are identical to [the tests in the main repo Travis](https://travis-ci.org/RaRe-Technologies/gensim/

**Comment by tmylk** [comment]
@souravsingh Maybe updating the version to 0.13.4.1 will fix the stalling build?

**Comment by souravsingh** [comment]
@jakirkham The recipe is getting an error due to missing ``unittest`` package. Is there a way to fix it?

**Comment by ericdill** [comment]
@souravsingh I think you need the `unittest2` package

**Comment by ericdill** [comment]
My mistake. unittest2 is a backport for py26 only.

File: recipes/gensim/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    - scipy >=0.7.0
+    - numpy >=1.3
+    - six >=1.5.0
+    - pyro4 >=4.27
+    - smart_open >=1.2.1
+
+test:
+  requires:
+    - nose
+    - unittest
```
Comment (jakirkham) [inline recipes/gensim/meta.yaml]: Please drop this. It is included in `python` already.

File: recipes/gensim/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/gensim/meta.yaml]: Then would recommend dropping everything except `python` and `setuptools` from `build`.

File: recipes/gensim/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+about:
+  home: http://github.com/RaRe-Technologies/gensim
+  license: LGPL 3.0
+  license_file: COPYING
+  license_family: LGPL
+  summary: 'A library for topic modelling and document indexing'
+
+  description: |
+    Gensim is a topic modelling, document indexing and similarity retrieval
+    library for Python 2.5+ and Python 3.3+. It is focussed towards Natural
```
Comment (tmylk) [inline recipes/gensim/meta.yaml]: Please update the description as 2.6 and 3.3 support was dropped

**Comment by souravsingh** [comment]
Closing the PR since it has become too cluttered.

**Pull request closed without merging**